arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -177,22 +211,32 @@ class ContentTableViewController: UITableViewController, CustomSearchHeaderViewD
         let storyboard = UIStoryboard(name: sample.storyboardName, bundle: .main)
         let controller = storyboard.instantiateInitialViewController()!
         controller.title = sample.name
-        let navController = UINavigationController(rootViewController: controller)
         
-        self.splitViewController?.showDetailViewController(navController, sender: self)
+        var presentingController:UIViewController? = self
+        if containsSearchResults {
+            //must use the controller presenting the results or else splitViewController will be nil
+            presentingController = presentingViewController
+        }
         
-        //add the button on the left on the detail view controller
-        if let splitViewController = self.view.window?.rootViewController as? UISplitViewController {
-            controller.navigationItem.leftBarButtonItem = splitViewController.displayModeButtonItem
-            controller.navigationItem.leftItemsSupplementBackButton = true
+        guard let splitViewController = presentingController?.splitViewController else {
+            return
         }
+            
+        let navController = UINavigationController(rootViewController: controller)
         
-        //create the info button and
-        //assign the readme url
+        //add the button on the left on the detail view controller
+        controller.navigationItem.leftBarButtonItem = splitViewController.displayModeButtonItem
+        controller.navigationItem.leftItemsSupplementBackButton = true
+        
+        //present the sample view controller
+        splitViewController.showDetailViewController(navController, sender: true)
```

Comment:
Please use `showDetailViewController(navController, sender: self)`. That should prevent you from needing to get a reference to the split view controller.

Reply:
The reference is still needed for `controller.navigationItem.leftBarButtonItem = splitViewController.displayModeButtonItem` but I'll change this to `presentingController.showDetailViewController(navController, sender: self)`. Using`self` directly isn't an option because of samples opening from `UISearchController` results.